stdlib/xml/etree/ElementTree.pyi

```diff
@@ -259,11 +260,27 @@ def fromstringlist(sequence: Sequence[str | bytes], parser: XMLParser | None = .
 _ElementFactory = Callable[[Any, Dict[Any, Any]], Element]
 
 class TreeBuilder:
-    def __init__(self, element_factory: _ElementFactory | None = ...) -> None: ...
+    if sys.version_info >= (3, 8):
+        def __init__(
+            self,
+            element_factory: _ElementFactory | None = ...,
+            *,
+            comment_factory: Callable[[str | None], Element] | None = ...,
```

Comment:
I don't think the argument of the factory ever be None. It is called from `comment(self, text: str)` via `self._handle_single`, which gives it the `text` as is.

Reply:
I'm sure there was a reason why I put it as `str | None`, but your analysis seems correct.

Reply:
Oh! I know why. It's because in `TreeBuilder.__init__`, it has:

```python
if comment_factory is None:
    comment_factory = Comment
```

`Comment` is a function, which is currently typed as `def Comment(text: str | None) -> Element: ...` (and has to be typed like that, since the default value of `text` is `None`.

So, I *think* it would be better to change the signature of `TreeBuilder.comment` to `comment(self, text: str | None)`.

Do you agree?

Reply:
It is fine to pass a function that can take `None`, even if it is never actually called with `None`. The type should reflect how the callback is actually used, not how the default value of the callback could be used.

Reply:
Ah, now I see what you mean. You're right. It is a bit weird to add a comment whose text is set to `None`, but because `Comment` supports it, I assume it does something reasonable and we should support it too.

Reply:
I will add relevant comments as well.